### ISSUE TITLE
The linker issue when using Carthage to build Siren library

### ISSUE THREAD

**Issue by yasuralian** [ArtSabintsev/Siren#125]
I'm using Xcode 8.2.1 and swift 3.0.2

```
func setupSiren() {
    let siren = Siren.sharedInstance
    siren.debugEnabled = true
    siren.alertType = .option
    siren.forceLanguageLocalization = .Japanese
    siren.checkVersion(checkType: .immediately)
}   
```

When I built my app with Carthage, I got this error.
<img width="885" alt="screen shot 2017-02-10 at 5 08 08 pm" src="https://cloud.githubusercontent.com/assets/11143310/22849711/897f6046-efb4-11e6-9bb0-20b0edb3eda7.png">

And this error description is almost same with #112  

Do you have any idea?

**Comment by yasuralian** [comment]
Temporally I just comment out a line like this:
`// siren.debugEnabled = true`

**Comment by ArtSabintsev** [comment]
What version of Siren are you using?

**Comment by yasuralian** [comment]
I'm using "1.1.3"

**Comment by ArtSabintsev** [comment]
Hmm, do you have a sample project you can send me? I don't use carthage myself, but I'm happy to help debug this issue.

**Comment by yasuralian** [comment]
I'm glad to hear you say that
please check [my repo](https://github.com/yasuralian/Sample_Siren)

**Comment by ArtSabintsev** [comment]
Working on it right now. Never seen this issue before.

**Comment by ArtSabintsev** [comment]
This is interesting. Apparently, Carthage has issues dealing with lazy variables.

 - https://github.com/xmartlabs/XLPagerTabStrip/issues/227#issuecomment-250235627
 - https://github.com/xmartlabs/Eureka/issues/693#issuecomment-252432122
 
Asking the community for help through other channels.

**Comment by calebd** [comment]
After some digging, this appears to be a Swift 3.0.x Whole Module Optimization bug. Switching precompiled  framework targets to `-0none` allows the build to complete successfully. It looks like this issue is fixed in the latest Xcode beta (8.3 beta 2).

**Comment by ArtSabintsev** [comment]
Thanks, @calebd!

**Comment by ArtSabintsev** [comment]
Closing this issue, but feel free to respond, @yasuralian.

**Comment by yasuralian** [comment]
Thank you @calebd @ArtSabintsev

**Comment by tatimagdalena** [comment]
I've seen this should be fixed on version 1.2.2, but I came across the same issue trying to use it from Carthage now.

It is giving me the same kind of error when trying to use differentiated Alerts for Major, Minor and Patch, and when trying to set appName...

**Comment by ArtSabintsev** [comment]
Can you make a new issue in Github, and link to this one? Thank you!
